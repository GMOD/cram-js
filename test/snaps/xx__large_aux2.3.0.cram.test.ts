import { testFile } from '../testUtil'
testFile('xx#large_aux2.3.0.cram')
