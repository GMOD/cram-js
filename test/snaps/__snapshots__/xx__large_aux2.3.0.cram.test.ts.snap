// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole xx#large_aux2.3.0.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 44,
      "_size": 18,
      "alignmentSpan": 0,
      "crc32": 3115796257,
      "landmarks": [
        0,
        120,
      ],
      "length": 189,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 44,
        "_size": 18,
        "alignmentSpan": 0,
        "crc32": 3115796257,
        "landmarks": [
          0,
          120,
        ],
        "length": 189,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              3,
              3,
              3,
              3,
              3,
              3,
              4,
              4,
              4,
              4,
            ],
            "numCodes": 10,
            "numLengths": 10,
            "symbols": [
              4,
              5,
              6,
              7,
              8,
              9,
              0,
              1,
              2,
              3,
            ],
          },
          "parametersBytes": 22,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {
        "aaC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "abC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "acC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "adC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "aeC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "afC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "agC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "ahC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "aiC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "ajC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
      },
      "tagIdsDictionary": [
        [
          "aaC",
        ],
        [
          "aaC",
          "abC",
        ],
        [
          "aaC",
          "abC",
          "acC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
          "agC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
          "agC",
          "ahC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
          "agC",
          "ahC",
          "aiC",
        ],
        [
          "aaC",
          "abC",
          "acC",
          "adC",
          "aeC",
          "afC",
          "agC",
          "ahC",
          "aiC",
          "ajC",
        ],
      ],
    },
    "containerHeader": {
      "_endPosition": 251,
      "_size": 18,
      "alignmentSpan": 1,
      "crc32": 1124434116,
      "landmarks": [
        525,
      ],
      "length": 719,
      "numBases": 10,
      "numBlocks": 6,
      "numLandmarks": 1,
      "numRecords": 10,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 251,
        "_size": 18,
        "alignmentSpan": 1,
        "crc32": 1124434116,
        "landmarks": [
          525,
        ],
        "length": 719,
        "numBases": 10,
        "numBlocks": 6,
        "numLandmarks": 1,
        "numRecords": 10,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a1",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
            },
            "templateSize": 0,
            "uniqueId": 782,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a2",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
            },
            "templateSize": 0,
            "uniqueId": 783,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a3",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
            },
            "templateSize": 0,
            "uniqueId": 784,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a4",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
            },
            "templateSize": 0,
            "uniqueId": 785,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a5",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
            },
            "templateSize": 0,
            "uniqueId": 786,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a6",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
            },
            "templateSize": 0,
            "uniqueId": 787,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a7",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
              "ag": 1,
            },
            "templateSize": 0,
            "uniqueId": 788,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a8",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
              "ag": 1,
              "ah": 1,
            },
            "templateSize": 0,
            "uniqueId": 789,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "a9",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
              "ag": 1,
              "ah": 1,
              "ai": 1,
            },
            "templateSize": 0,
            "uniqueId": 790,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 1,
            "qualityScores": [
              2,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 1,
            "readName": "aA",
            "sequenceId": 0,
            "tags": {
              "aa": 1,
              "ab": 1,
              "ac": 1,
              "ad": 1,
              "ae": 1,
              "af": 1,
              "ag": 1,
              "ah": 1,
              "ai": 1,
              "aj": 1,
            },
            "templateSize": 0,
            "uniqueId": 791,
          },
        ],
        "header": {
          "_endPosition": 816,
          "_size": 35,
          "compressedSize": 31,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            1,
            10,
            0,
            4,
            3,
            1,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            127,
            197,
            98,
            112,
            231,
            167,
            15,
            168,
            26,
            89,
            53,
            183,
            46,
            172,
            190,
            41,
          ],
          "contentId": 0,
          "contentPosition": 781,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 133776389,
          "parsedContent": {
            "_endPosition": 282,
            "_size": 31,
            "contentIds": [
              1,
              11,
              12,
            ],
            "md5": [
              127,
              197,
              98,
              112,
              231,
              167,
              15,
              168,
              26,
              89,
              53,
              183,
              46,
              172,
              190,
              41,
            ],
            "numBlocks": 4,
            "numContentIds": 3,
            "numRecords": 10,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 1,
            "refSeqStart": 1,
          },
          "uncompressedSize": 31,
        },
      },
    ],
  },
]
`;
